src/acvp_kdf135_ssh.c

```diff
@@ -0,0 +1,386 @@
+    }
+    json_free_serialized_string(json_result);
+
+    return ACVP_SUCCESS;
+}
+
+/*
+ * After the test case has been processed by the DUT, the results
+ * need to be JSON formated to be included in the vector set results
+ * file that will be uploaded to the server.  This routine handles
+ * the JSON processing for a single test case.
+ */
+static ACVP_RESULT acvp_kdf135_ssh_output_tc(ACVP_CTX *ctx, ACVP_KDF135_SSH_TC *stc, JSON_Object *tc_rsp)
+{
+    char *tmp;
+
+    tmp = calloc(1, ACVP_KDF135_SSH_MSG_MAX);
+    if (!tmp) {
+        ACVP_LOG_ERR("Unable to malloc in acvp_kdf135_ssh_output_tc");
+        return ACVP_MALLOC_FAIL;
+    }
+
+    json_object_set_string(tc_rsp, "csInitIV", tmp);
+
+    tmp = calloc(1, ACVP_KDF135_SSH_MSG_MAX);
+    if (!tmp) {
+        ACVP_LOG_ERR("Unable to malloc in acvp_kdf135_ssh_output_tc");
+        return ACVP_MALLOC_FAIL;
+    }
+
+    json_object_set_string(tc_rsp, "scInitIV", tmp);
+
+    tmp = calloc(1, ACVP_KDF135_SSH_MSG_MAX);
+    if (!tmp) {
+        ACVP_LOG_ERR("Unable to malloc in acvp_kdf135_ssh_output_tc");
+        return ACVP_MALLOC_FAIL;
+    }
+
+    json_object_set_string(tc_rsp, "csEKey", tmp);
+
+    tmp = calloc(1, ACVP_KDF135_SSH_MSG_MAX);
+    if (!tmp) {
+        ACVP_LOG_ERR("Unable to malloc in acvp_kdf135_ssh_output_tc");
+        return ACVP_MALLOC_FAIL;
+    }
+
+    json_object_set_string(tc_rsp, "scEKey", tmp);
+
+    tmp = calloc(1, ACVP_KDF135_SSH_MSG_MAX);
+    if (!tmp) {
+        ACVP_LOG_ERR("Unable to malloc in acvp_kdf135_ssh_output_tc");
+        return ACVP_MALLOC_FAIL;
+    }
+
+    json_object_set_string(tc_rsp, "csIKey", tmp);
+
+    tmp = calloc(1, ACVP_KDF135_SSH_MSG_MAX);
+    if (!tmp) {
+        ACVP_LOG_ERR("Unable to malloc in acvp_kdf135_ssh_output_tc");
+        return ACVP_MALLOC_FAIL;
+    }
+
+    json_object_set_string(tc_rsp, "scIKey", tmp);
+    return ACVP_SUCCESS;
+}
+
+static ACVP_RESULT acvp_kdf135_ssh_init_tc(ACVP_CTX *ctx,
+                                    ACVP_KDF135_SSH_TC *stc,
+                                    unsigned int tc_id,
+                                    ACVP_CIPHER alg_id,
+                                    unsigned int  sha_type,
+                                    unsigned int  sh_sec_len,
+                                    unsigned int  iv_len,
+                                    unsigned int  key_len,
+                                    const unsigned char *shared_sec_k,
+                                    const unsigned char *hash_h,
+                                    unsigned int hash_len,
+                                    const unsigned char *session_id,
+                                    unsigned int session_len)
+{
+    memset(stc, 0x0, sizeof(ACVP_KDF135_SSH_TC));
+
+    stc->shared_sec_k = calloc(1, ACVP_KDF135_SSH_MSG_MAX);
+    if (!stc->shared_sec_k) return ACVP_MALLOC_FAIL;
+    stc->hash_h = calloc(1, ACVP_KDF135_SSH_MSG_MAX);
+    if (!stc->hash_h) return ACVP_MALLOC_FAIL;
```

Comment:
Hey Norm, I see where shared_sec_k and hash_h are given space but I don't see where the information for them is put in the buffer you created.